apps/web/src/components/profile/my-profile-currency.tsx

```diff
@@ -1,92 +1,36 @@
-import { DEFAULT_CURRENCY } from '@algomart/schemas'
-import { useRouter } from 'next/router'
 import useTranslation from 'next-translate/useTranslation'
-import { FormEvent, useCallback, useMemo, useState } from 'react'
-import { ExtractError } from 'validator-fns'
+import { useCallback, useState } from 'react'
 
 import common from './my-profile-common.module.css'
 import css from './my-profile-language.module.css'
 
 import { Currency } from '@/components/auth-inputs/auth-inputs'
-import Button from '@/components/button'
 import Heading from '@/components/heading'
-import { useAuth } from '@/contexts/auth-context'
-import { useCurrency } from '@/hooks/use-currency'
-import { AuthService } from '@/services/auth-service'
-import { validateCurrency } from '@/utils/auth-validation'
+import { useCurrency } from '@/contexts/currency-context'
 
 export default function MyProfileCurrency() {
-  const { user, reloadProfile } = useAuth()
-  const [formErrors, setFormErrors] = useState<ExtractError<typeof validate>>()
-  const [isEditing, setIsEditing] = useState<boolean>(false)
-  const [currency, setCurrency] = useState<string>(useCurrency())
-  const [loading, setLoading] = useState<boolean>(false)
-  const [updateError, setUpdateError] = useState<string>('')
-  const [updateSuccess, setUpdateSuccess] = useState<boolean>(false)
+  const { currency, updateCurrency } = useCurrency()
+  const [error, setError] = useState('')
+  const [updateSuccess, setUpdateSuccess] = useState(false)
   const { t } = useTranslation()
-  const router = useRouter()
 
-  const validate = useMemo(() => validateCurrency(t), [t])
-
-  const handleUpdateCurrency = useCallback(
-    async (event: FormEvent<HTMLFormElement>) => {
-      event.preventDefault()
-      setLoading(true)
-      setUpdateError('')
+  const handleCurrencyChange = useCallback(
+    (currency) => {
+      setError('')
       setUpdateSuccess(false)
-
-      // Validate form body
-      const body = {
-        currency,
-      }
-      const bodyValidation = await validate(body)
-      if (bodyValidation.state === 'invalid') {
-        setFormErrors(bodyValidation.errors)
-        setLoading(false)
-        return
-      }
-
-      // Update language
-      const updateCurrency = await AuthService.instance.updateCurrency(
-        body.currency
-      )
-      if (!updateCurrency) {
-        setUpdateError(t('common:statuses.An Error has Occurred'))
-        setLoading(false)
-        return
-      }
-
-      await reloadProfile()
-      setLoading(false)
-      setIsEditing(false)
-      setFormErrors({})
-      setUpdateError('')
-      setUpdateSuccess(true)
-
-      router.push(
-        { pathname: router.pathname, query: router.query },
-        router.asPath
-      )
-
-      return
+      updateCurrency(currency).then((success) => {
+        if (success) {
+          setError('')
+          setUpdateSuccess(true)
+          setTimeout(() => setUpdateSuccess(false), 3000)
```

Comment:
Can we add a comment here? Not sure I follow what's going on/why this is needed.

Reply:
you bet, I'll also try to come up with a better name for this. Took me a minute to figure out what the heck it was doing too.

Basically it's whether or not to show a success message after saving. So we show the message and then hide after 3s.